internal/pkg/deploy/cloudformation/pipeline.go

```diff
@@ -65,13 +70,30 @@ func (cf CloudFormation) CreatePipeline(in *deploy.CreatePipelineInput, bucketNa
 	if err = cf.codeStarClient.WaitUntilConnectionStatusAvailable(ctx, output[connectionARNKey]); err != nil {
 		return err
 	}
-	if err = cf.cpClient.RetryStageExecution(in.Name, sourceStage); err != nil {
+
+	pipelineResourceName, err := cf.pipelinePhysicalResourceID(stackConfig.StackName())
+	if err != nil {
+		return err
+	}
+	if err = cf.cpClient.RetryStageExecution(pipelineResourceName, sourceStage); err != nil {
 		return err
 	}
-
 	return nil
 }
 
+func (cf CloudFormation) pipelinePhysicalResourceID(stackName string) (string, error) {
+	stackResources, err := cf.cfnClient.StackResources(stackName)
+	if err != nil {
+		return "", err
+	}
+	for _, resource := range stackResources {
+		if aws.StringValue(resource.LogicalResourceId) == cfnLogicalResourceIDPipeline && aws.StringValue(resource.ResourceType) == cfnResourceTypePipeline {
+			return aws.StringValue(resource.PhysicalResourceId), nil
+		}
+	}
+	return "", fmt.Errorf("cannot find a resource in stack %s with logical ID \"%s\" of type \"%s\"", stackName, cfnLogicalResourceIDPipeline, cfnResourceTypePipeline)
```

Comment:
this is basically a new error though instead of an error that's wrapping a parent error. It would have been "error wrapping CreatePipeline: find resource in stack pipeline-app-pipeline with logical ID pipeline of type CodePipeline::Pipeline" 🤔  reads a little weird without "cannot"